src/components/views/DownloadProgressModal.vue

```diff
@@ -23,28 +25,45 @@ function closeModal() {
             <div class='modal-content'>
                 <div class='notification is-info'>
 
-                    <h3 v-if="$store.getters['download/currentDownload'].downloadProgress < 100" class='title'>
+                    <h3 v-if="DownloadUtils.statusIsDownloadOrExtract($store.getters['download/currentDownload'].status)" class='title'>
                         Downloading {{$store.getters['download/currentDownload'].modName}}
                     </h3>
-                    <h3 v-else class='title'>
+                    <h3 v-else-if="$store.getters['download/currentDownload'].status === DownloadStatusEnum.INSTALLING" class='title'>
                         Installing {{$store.getters['download/currentDownload'].modName}}
                     </h3>
 
-                    <p>
+
+                    <p v-if="$store.getters['download/currentDownload'].status === DownloadStatusEnum.DOWNLOADING">
+                        <font-awesome-icon icon="fa-regular fa-download" />
                         Downloading: {{$store.getters['download/currentDownload'].downloadProgress}}% of
                         {{FileUtils.humanReadableSize($store.getters['download/currentDownload'].totalDownloadSize)}}
                     </p>
 
+                    <p v-else-if="$store.getters['download/currentDownload'].status === DownloadStatusEnum.EXTRACTING || $store.getters['download/currentDownload'].status === DownloadStatusEnum.EXTRACTED">
+                        <font-awesome-icon icon="fa-regular fa-box-open" />
+                        Extracting: {{$store.getters['download/currentDownload'].downloadProgress}}% of
```

Comment:
I like how the download monitor show this, i.e. "Downloading: mod name" and "Extracting: mod name" alternating on one row and "X% of Y MB" on a separate row. Here the mod name is on the header row and extracting/installing is combined with percentage/total size. Feels a bit jumpy and confusing.

I'm not exactly sure how I'd remedy this so we could have a separate task for rethinking how the data is shown in the modal. Currently it shows that more pieces have been bolted on top of the old pieces of this Frankenstein's Modal too many times.

Reply:
Created a task for this